File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
      int max=0;
        int buy=prices[0];
        int sell=prices[0];
        
        for(int i=0;i<prices.length;i++){
            buy = Math.min(buy, prices[i]); // 1
            sell = Math.max(buy, prices[i]); //6
            
            max = Math.max(max, (sell-buy));
        }
        return max;
    }
}

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.go
func maxProfit(prices []int) int {
    if len(prices)==0{
        return 0
    }
    
    max := 0
    sell := prices[0]
    buy := prices[0]
    
    for i:=0; i<len(prices);i++{
        buy = int(math.Min(float64(buy),float64(prices[i])))
        sell = int(math.Max(float64(buy), float64(prices[i])))
        max = int(math.Max(float64(max), float64(sell-buy)))
    }
    
    return max
    
}